### PR TITLE
[legacy]disabler && Grim NoXZ Velocity && hypixel fu**er

### PR DESCRIPTION
**Pull request by Auza2008**
lessflag is just an example. the rest of the disabler needs to be done by someone else. LOL

GrimVelocity is skid from other client.(Requires the server to have a bad configuration)

Hypixel Fuc*er bypass

### CODE REVIEW

**Comment by CzechHek** [comment]
The code is weird.
Can everyone stop messing with dynamic values indentions, I bet that if AI would read the text, it would understand that conditionally shown values are indented and it would indent new values as well.

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Fucker.kt [hunk 1]
```diff
@@ -46,17 +47,18 @@ object Fucker : Module("Fucker", ModuleCategory.WORLD) {
     private val range by FloatValue("Range", 5F, 1F..7F)
 
     private val action by ListValue("Action", arrayOf("Destroy", "Use"), "Destroy")
-    private val surroundings by BoolValue("Surroundings", true)
-        private val instant by BoolValue("Instant", false) { action == "Destroy" || surroundings }
+    private var surroundings by BoolValue("Surroundings", true)
+    private val hypixelMode by BoolValue("Hypixel-Bypass", false)
```
Comment (CzechHek) [inline src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Fucker.kt:51]: Ah yes jut put always shown value above Instant value, which depends on two values above it, why not.

Comment (Auza2008) [reply src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Fucker.kt:51]: my bad

Comment (Auza2008) [reply src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Fucker.kt:51]: idea code auto clean

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/combat/Velocity.kt [hunk 2]
```diff
@@ -249,6 +255,26 @@ object Velocity : Module("Velocity", ModuleCategory.COMBAT) {
...
+                        val target = LiquidBounce.targetManager.getNearByEntity(3f)
+                        repeat(12) {
+                            mc.netHandler.addToSendQueue(C0FPacketConfirmTransaction())
+                            mc.thePlayer.sendQueue.addToSendQueue(
+                                C02PacketUseEntity(
+                                    target,
+                                    C02PacketUseEntity.Action.ATTACK
+                                )
+                            )
+                            mc.thePlayer.sendQueue.addToSendQueue(C0APacketAnimation())
```
Comment (CzechHek) [inline src/main/java/net/ccbluex/liquidbounce/features/module/modules/combat/Velocity.kt]: sendPacket

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/exploit/disablermodes/other/LessFlag.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                event.cancelEvent()
+                mc.netHandler.networkManager.sendPacket(
+                    C06PacketPlayerPosLook(
+                        packetS08.x,
+                        packetS08.y,
+                        packetS08.z,
+                        packetS08.getYaw(),
+                        packetS08.getPitch(),
+                        true
+                    )
```
Comment (CzechHek) [inline src/main/java/net/ccbluex/liquidbounce/features/module/modules/exploit/disablermodes/other/LessFlag.kt]: sendPacket

packet.yaw
packet.pitch

why always on ground

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Fucker.kt (hunk 1 above)
Comment (CzechHek) [reply src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Fucker.kt:51]: By putting 1 line space between hypixel-bypass and instant, the instant is still below, separated by totally independent value?

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Fucker.kt [hunk 4]
```diff
@@ -134,6 +137,21 @@ object Fucker : Module("Fucker", ModuleCategory.WORLD) {
             }
         }
 
+        val block = Block.getIdFromBlock(getBlock(currentPos)) == targetId
```
Comment (CzechHek) [inline src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Fucker.kt]: What is this variable naming?

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/exploit/disablermodes/other/LessFlag.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import net.minecraft.network.play.server.S08PacketPlayerPosLook
+
+object LessFlag : DisablerMode("LessFlag") {
+    override fun onPacket(event: PacketEvent) {
+        val packet = event.packet
+        if (packet is S08PacketPlayerPosLook) {
+            val packetS08 = packet
+            val x = packetS08.x - mc.thePlayer.posX
+            val y = packetS08.y - mc.thePlayer.posY
+            val z = packetS08.z - mc.thePlayer.posZ
```
Comment (CzechHek) [inline src/main/java/net/ccbluex/liquidbounce/features/module/modules/exploit/disablermodes/other/LessFlag.kt]: What is this variable naming 2?

File: src/main/java/net/ccbluex/liquidbounce/manager/TargetManager.kt [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        attackedEntityList.clear()
+    }
+
+    fun getNearByEntity(radius: Float): EntityLivingBase? {
+        return try {
+            mc.theWorld.loadedEntityList.filter {
+                mc.thePlayer.getDistanceToEntity(it) < radius && EntityUtils.isSelected(
+                    it, true
+                )
+            }.sortedBy { it.getDistanceToEntity(mc.thePlayer) }[0] as EntityLivingBase?
```
Comment (CzechHek) [inline src/main/java/net/ccbluex/liquidbounce/manager/TargetManager.kt]: .minBy { }

+ calculates distances twice, could store them in pairs
+ nearby?
+ why try catch, just make it null safe

File: src/main/java/net/ccbluex/liquidbounce/manager/TargetManager.kt [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        }
+    }
+
+    @EventTarget
+    fun onAttack(event: AttackEvent) {
+        val target = event.targetEntity
+
+        if (target is EntityLivingBase && EntityUtils.isSelected(target, true)) {
+            this.target = target
+            if (!attackedEntityList.contains(target)) attackedEntityList.add(target)
```
Comment (CzechHek) [inline src/main/java/net/ccbluex/liquidbounce/manager/TargetManager.kt]: target !in
list += target

File: src/main/java/net/ccbluex/liquidbounce/manager/TargetManager.kt [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+            }
+        }
+
+        if (target != null) {
+            if (mc.thePlayer.getDistanceToEntity(target) > 7 || !inCombat || target!!.isDead) {
+                target = null
+            }
+        }
+
+        attackedEntityList.map { it }.forEach {
```
Comment (CzechHek) [inline src/main/java/net/ccbluex/liquidbounce/manager/TargetManager.kt]: list.map { it }???

also how many times do you calculate distances

**Review by CzechHek — CHANGES_REQUESTED** [review]

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/combat/Velocity.kt [hunk 9]
```diff
@@ -249,6 +255,26 @@ object Velocity : Module("Velocity", ModuleCategory.COMBAT) {
...
+                        }
+                        event.cancelEvent()
+                        mc.thePlayer.motionY = packet.motionY.toDouble() / 8000.0
+                        mc.thePlayer.motionX *= 0
+                        mc.thePlayer.motionZ *= 0
```
Comment (CzechHek) [inline src/main/java/net/ccbluex/liquidbounce/features/module/modules/combat/Velocity.kt]: mc.thePlayer.motionY = packet.realMotionY

mc.thePlayer.stopXZ()

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/exploit/disablermodes/other/LessFlag.kt (hunk 3 above)
Comment (Auza2008) [reply src/main/java/net/ccbluex/liquidbounce/features/module/modules/exploit/disablermodes/other/LessFlag.kt]: just an example

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Fucker.kt (hunk 1 above)
Comment (Auza2008) [reply src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Fucker.kt:51]: okok, My translation software misinterpreted your meaning, I thought you were talking about sequencing

**Comment by Auza2008** [comment]
Ok, I modified the targetmanager the way you did, but I'm not sure I did it right!

**Comment by gabrielvicenteYT** [comment]
The code is very low quality, this is NOT FDPClient!

**Comment by Auza2008** [comment]
> The code is very low quality, this is NOT FDPClient!

I'm sorry, but that's all my powers allow me to do, and if you're upset, you can absolutely rewrite it.

**Comment by EclipsesDev** [comment]
have you atleast tested the bypass etc?

**Comment by Auza2008** [comment]
> have you atleast tested the bypass etc?

At least grim velocity works on some servers with bad configurations.

**Comment by 1zun4** [comment]
Seems OK to me, but I would advise removing the TargetTracker from the PR, as the only function used is `getNearByEntity` and does not require the functionality of the tracker itself.

If this has been changed, I will accept this PR :)

**Comment by gabrielvicenteYT** [comment]
> Seems OK to me, but I would advise removing the TargetTracker from the PR, as the only function used is `getNearByEntity` and does not require the functionality of the tracker itself.
> 
> If this has been changed, I will accept this PR :)

The code is skidded, which is a no-no

**Comment by Cosmic-SC** [comment]
> > Seems OK to me, but I would advise removing the TargetTracker from the PR, as the only function used is `getNearByEntity` and does not require the functionality of the tracker itself.
> > If this has been changed, I will accept this PR :)
> 
> The code is skidded, which is a no-no

yep that less flag disabler probably makes u flag more

**Comment by CzechHek** [comment]
If all other skidded PRs weren't merged, why merge this?

**Comment by 1zun4** [comment]
> If all other skidded PRs weren't merged, why merge this?

Seems to be basic code, just because someone else has it should not mean we should always reject something. It seems he knew what he was doing and was able to adapt to my request, so I am not too worried about it being taken from somewhere else. Especially grimcombat is just attacking the entity, thats it, there is nothing to skid, no exact value, no super complex methods, just attack on knockback and cancel with XZ.
For the LessFlag it is the same, but its horrible code. It may be auto-translated from Java, but in that case it is pretty basic as well (and probably useless?).

I cannot always tell if something is stolen or not, but I cannot always assume that it is. Also, it is legacy that I do not work on, but it annoys me when PR stays open forever. If the code is taken from somewhere private you can tell me and we will revert the changes.

Edit: Nvm, just read the entry commit message "GrimVelocity is skid from other client.(Requires the server to have a bad configuration)". Welp, thats a revert.,

**Pull request merged**